src/osx/cocoa/dataview.mm

```diff
@@ -2508,6 +2508,19 @@ void UpdateWithRow(int row)
             byExtendingSelection:GetDataViewCtrl()->HasFlag(wxDV_MULTIPLE) ? YES : NO];
 }
 
+void wxCocoaDataViewControl::Select(const wxDataViewItemArray& items)
+{
+    NSMutableIndexSet *selection = [[NSMutableIndexSet alloc] init];
```

Comment:
We don't need to release this?

Reply:
Sorry, I didn't realize wx doesn't use [ARC](https://en.wikipedia.org/wiki/Automatic_Reference_Counting). Yes, it should be released; I just didn't write manually refcounting code in a long time. Will fix.

I got to ask, though: if wx now requires a sufficiently modern macOS compiler to have C++11 (nice!), why doesn't it use ARC too? As far as I can tell, they're both available since Xcode 4.2.

Reply:
Sorry, no idea. Perhaps because the existing code would need to be updated to work with ARC? @csomor do you think we should enable/require ARC for wxMac?

Reply:
@vslavik long time ago I think one of the models was contagious, ie when writing plug-ins your host app had to use it too, but I guess this was the garbage collection. So with ARC it's probably just the question of mixing header files. Can this be an issue?

Reply:
@csomor @vadz GC was contagious but ARC is purely compile-time, so mixing & matching is possible, and wx headers are already ARC-neutral (if that's what you mean). 

I take it there are no deeper reasons, so you won't mind if I take a look at ARC conversion?

Reply:
If this doesn't require changing our current minimum requirements (i.e. "SDK 10.11, Xcode 7.2.1 or greater (runs under 10.10.5 and higher)", see `docs/doxygen/mainpages/platdetails.h`), then definitely not (if it does, I might still not mind, but if it doesn't, there is nothing to even discuss, AFAICS).

Reply:
> @csomor @vadz GC was contagious but ARC is purely compile-time, so mixing & matching is possible, and wx headers are already ARC-neutral (if that's what you mean).

Thanks a lot for clarifying

I meant it the other way round, say eg I'm developing an Adobe Photoshop plug-in, can this be a problem to include for a wx-based plug-in

> I take it there are no deeper reasons, so you won't mind if I take a look at ARC conversion?

that would be perfect, yes, thank you

Reply:
> I meant it the other way round, say eg I'm developing an Adobe Photoshop plug-in, can this be a problem to include for a wx-based plug-in

I don't think it can — ARC really is just per-compilation-unit processing. It's nothing other than the compiler, instead of a human, inserting the retain/release calls. There could be some issues with newly created objects (where Obj-C function name is used by ARC to enforce what was previously a convention), but I don't think that applies to wx code and certainly not to its public API.

Reply:
ok, then I'm all for it, thanks